app/models/metadatum.rb

```diff
@@ -0,0 +1,101 @@
+class Metadatum < ApplicationRecord
+  belongs_to :sample
+
+  # Types: 0 for string, 1 for number, 2 for lists of values (as string)
+  KEY_TO_TYPE = { unique_id: 0, sample_type: 0, nucleotide_type: 0, collection_date: 0, collection_location: 0, collected_by: 0, age: 1, gender: 0, race: 0, primary_diagnosis: 0, antibiotic_administered: 0, admission_date: 0, admission_type: 0, discharge_date: 0, discharge_type: 0, immunocomp: 0, other_infections: 0, comorbitity: 0, known_organism: 0, infection_class: 0, detection_method: 0, library_prep: 0, sequencer: 0, rna_dna_input: 1, library_prep_batch: 0, extraction_batch: 0 }.freeze
```

Comment:
I would suggestion formatting this one per line. Makes it easier to read.

Reply:
Would also suggest using enums for readability: https://api.rubyonrails.org/v5.2.1/classes/ActiveRecord/Enum.html

Reply:
i think you should also define constant for type like METADATA_TYPE_INTEGER , etc. agreed with Tiago on one per line